django/db/models/sql/query.py

```diff
@@ -1940,7 +1948,7 @@ def set_group_by(self, allow_aliases=True):
                     warnings.warn(msg, category=RemovedInDjango40Warning)
                     group_by_cols = annotation.get_group_by_cols()
                 else:
-                    if not allow_aliases:
+                    if not allow_aliases or (alias and alias in self._get_alias_map_tables_column_names()):
```

Comment:
Now that I think of it you can likely avoid a full set materialization by passing `alias` to the function and having it return as soon as it has a match.

e.g.

```python
def _column_name_conflict(self, alias):
    models = {}
    for join in list(self.alias_map.values())[1:]:
        model = join.join_field.related_model
        if model in models:
            continue
        for field in model._meta.local_concrete_fields:
            if field.column == alias:
                return True
        models.add(model)
    return False

Reply:
Thanks again.
Yes, the test pass with `related_model` only.
I've replaced your suggested function.